File: backend/src/HealthCore.Domain/Entities/Prescription.cs
using HealthCore.Domain.Common;

namespace HealthCore.Domain.Entities;

public class Prescription : BaseEntity
{
    public Guid PatientId { get; set; }
    public Guid DoctorId { get; set; }
    public Guid MedicalConsultationId { get; set; }

    public string Medication { get; set; } = string.Empty;
    public string Dosage { get; set; } = string.Empty;
    public string Frequency { get; set; } = string.Empty;
    public string Duration { get; set; } = string.Empty;
    public string? Instructions { get; set; }

    public Patient Patient { get; set; } = null!;
    public Doctor Doctor { get; set; } = null!;
    public MedicalConsultation MedicalConsultation { get; set; } = null!;
}

File: frontend/src/types/prescription.ts
export interface Prescription {
  id: string
  patientId: string
  patientName: string
  doctorId: string
  doctorName: string
  medicalConsultationId: string
  medication: string
  dosage: string
  frequency: string
  duration: string
  instructions?: string | null
  createdAt: string
}

export interface CreatePrescriptionFormData {
  patientId: string
  doctorId: string
  medicalConsultationId: string
  medication: string
  dosage: string
  frequency: string
  duration: string
  instructions: string
}

export interface UpdatePrescriptionFormData {
  doctorId: string
  medicalConsultationId: string
  medication: string
  dosage: string
  frequency: string
  duration: string
  instructions: string
}
